ivadomed/loader/patch_filter.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+
+
+class PatchFilter(object):
+    """Filter 2D patches from dataset.
+
+    If a patch does not meet certain conditions, it is discarded from the dataset at training time.
+
+    Args:
+        filter_empty_mask (bool): If True, 2D patches where all voxel labels are zeros are discarded at training time.
+        filter_absent_class (bool): If True, 2D patches where all voxel labels are zero for one or more classes are
+            discarded at training time.
+        filter_empty_input (bool): If True, 2D patches where all voxel intensities are zeros are discarded
+            at training time.
+        is_train (bool): Indicates if at training time.
+
+    Attributes:
+        filter_empty_mask (bool): If True, 2D patches where all voxel labels are zeros are discarded at training time.
+            Default: False.
+        filter_absent_class (bool): If True, 2D patches where all voxel labels are zero for one or more classes are
+            discarded at training time. Default: False.
+        filter_empty_input (bool): If True, 2D patches where all voxel intensities are zeros are discarded
+            at training time. Default: False.
+        is_train (bool): Indicates if at training time.
+
+    """
+
+    def __init__(self, filter_empty_mask=False,
+                 filter_absent_class=False,
+                 filter_empty_input=False,
+                 is_train=False):
+        self.filter_empty_mask = filter_empty_mask
+        self.filter_absent_class = filter_absent_class
+        self.filter_empty_input = filter_empty_input
+        self.is_train = is_train
```

Comment:
Also these ones as well.  `is_` prefix is pretty good at indicating binaries. but `filter_` flag might be less intuitive new devs

Reply:
Agreed, done in 0761115.